lib/lita/handlers/api/market.rb

```diff
@@ -14,40 +14,41 @@ def initialize(robot)
 
         http.get 'market/limit_orders', :limit_orders
         http.post 'market/limit_orders', :place_limit_order
-        http.post 'market/market_orders', :place_market_order
+        http.post 'market/execute_transaction', :execute_transaction
 
         def limit_orders(request, response)
           return respond_not_authorized(response) unless authorized?(request)
           orders = market_manager.orders
           respond(response, limit_orders: orders)
         end
 
-        def place_limit_order(request, response)
+        def execute_transaction(request, response)
           return respond_not_authorized(response) unless authorized?(request)
           user = current_user(request)
-          order = limit_order_for_user(user)
           if user
-            if winning_list.include?(user.mention_name) && market_manager.add_limit_order(order)
-              respond(response, success: true, order: order)
-            else
+            executed_orders = market_manager.execute_transaction
+            unless executed_orders
               response.status = 403
-              respond(response, status: 403, message: 'Can not place order')
+              respond(response, status: 403, message: 'Any transaction possible')
             end
-          else
-            response.status = 404
-            respond(response, status: 404, message: 'Error in parameters')
+            respond(response, success: true, orders: executed_orders.to_json)
           end
         end
 
-        def place_market_order(request, response)
+        def place_limit_order(request, response)
```

Comment:
Cyclomatic complexity for place_limit_order is too high. [9/6]<br>Method has too many lines. [18/15]<br>Perceived complexity for place_limit_order is too high. [11/7]